RIOT-OS/RIOT
# cpu/esp32: build the SDK bootloader from source

### Contribution description

As follow-up to PR #17043 this PR provides the ESP32 bootloader as module which is compiled from SDK sources.

Up to now, there were four precompiled versions of bootloaders (`bootloader.bin`) for different settings of logging and colored logging but with a fix size of flash and a fix UART baudrate. To change these parameters, the bootloader has to be recompiled. However there are more settings that affect the bootloader build that might be relevant to the app or final board that uses this bootloader.

This PR builds the `bootloader.bin` file from the ESP32 SDK sourc as a module `esp_bootloader`. It doesn't generate any object code for the application and only produces a `bootloader.bin` file set to the `BOOTLOADER_BIN` make variable for the `esptool.inc.mk` to flash.

This PR includes #17455 for the moment to be compilable (commits fded044ebe62ae4158b4ae4b482b4279d217dcf8 b63912314f6cc9fcec6e59476d599196c01319d6 8a840850883ba05ad3ab7732e9f3423ed99de7fb 3b4c5e99070dd70ee65cd7c7621efb1fb3e599a3 0c56a2de5375b8817788a3f65897c67d38ac0c29).

### Testing procedure

Use command
```
USEMODULE='esp_log_startup esp_log_colored' BAUD=57600  RIOT_TERMINAL=picocom \
BOARD=esp32-wroom-32 make -C tests/shell flash term 
```
and press RESET. The second-stage output should be seen in color. The first-stage bootloader and the application output is seen as garbage since they use a 115200 as baudrate.

### Issues/PRs references

Depends on PR #17455 
Follow-up to PR #17043
